tests/unit/calmBull.test.ts

```diff
@@ -0,0 +1,142 @@
+import { describe, expect, test } from '@jest/globals'
+import { BigNumber, Contract } from 'ethers'
+import { QUOTER } from '../../constants/address'
+import { BIG_ONE, DEFAULT_SLIPPAGE, WETH_DECIMALS_DIFF } from '../../constants/numbers'
+import { Quoter } from '../../types/contracts'
+import quoterAbi from '../../abis/quoter.json'
+import { getLeverageRebalanceDetails } from '../../utils/calmBull'
+import * as quoterFns from '../../utils/quoter'
+import '../../utils/math'
+
+const quoter = new Contract(QUOTER, quoterAbi) as Quoter
+
+describe('CalmBull: Leverage Rebalance', () => {
+  const crabBalance = BIG_ONE.mul(200)
+  const loanCollat = BIG_ONE.mul(200)
+  const loanDebt = BIG_ONE.mul(100_000).div(WETH_DECIMALS_DIFF)
+  const crUpper = BigNumber.from('3000000') // 3
+  const crLower = BigNumber.from('1200000') // 1.2
+  const deltaUpper = BIG_ONE.mul(12).div(10) // 1.2
+  const deltaLower = BIG_ONE.mul(9).div(10) //.9
+
+  function mockQuoterFunctions(ethPrice: BigNumber) {
+    jest
+      .spyOn(quoterFns, 'quoteExactIn')
+      .mockImplementation((quoter: Quoter, tokenIn: string, tokenOut: string, amountIn: BigNumber, poolFee: number) => {
+        return Promise.resolve(amountIn.wdiv(ethPrice))
+      })
+
+    jest
+      .spyOn(quoterFns, 'quoteExactOut')
+      .mockImplementation(
+        (quoter: Quoter, tokenIn: string, tokenOut: string, amountOut: BigNumber, poolFee: number) => {
+          return Promise.resolve(amountOut.wdiv(ethPrice))
+        },
+      )
+  }
+
+  /**
+   * Initial values
+   * ETH price = 1000
+   * crabPrice = 500
+   * sqth price = .1
+   * CR = 2
+   * delta = 1
+   */
+  test('Rebalance when ETH price go down', async () => {
```

Comment:
Leverage rebalance should only be used when crab changes value relative to debt, so think the eth price change doesn't matter as much for this test - the fact we leverage rebalance is really that crab reduces in value relative to debt, so probably best to make that the explicit name of what test is doing

Reply:
basically delta being different from 1 can be fixed by full or leverage rebalances, as long as CR is respected

delta only changes when the crab vs debt value changes, but becomes larger or smaller depending on eth price changes as well (but only changes if there is a change in crab value, which in practice will be 100% of the time :) )

CR changes when eth price changes (which can also have an impact on the crab value to be clear as ETH price changes oSQTH price assuming vol constant)